### PR TITLE
button component and unit tests

### PR DESCRIPTION
**Pull request by F-Gulmez**

### CODE REVIEW

File: Frontend/src/components/Buttons.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    const [count, setCount] = useState(0);
+    const incrementCounter=() => {
+        setCount(count + 1)
+    };
+    const decrementCounter=() => {
+        setCount(count - 1)
+    }
+    return (
+        <div >
+        <h1 data-testid="counter-test">{count}</h1>
```
Comment (yulingyou) [inline Frontend/src/components/Buttons.jsx:12]: Would `data-testid` be visible in the production HTML?

File: Frontend/tests/components/buttons.test.jsx
Comment (yulingyou) [inline Frontend/tests/components/buttons.test.jsx:1]: Additional test cases to ensure successful transition from positive to negative position on the gauge counter. We suggest to add two additional test cases:
1. test("Test that When User clicks Power Up Button once  and Power Down Button twice Counter number changes to -1”
2. test("Test that When User clicks Power Down Button once  and Power Up Button twice Counter number changes to 1”
See the code suggestion below

File: Frontend/tests/components/buttons.test.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    const powerUpButton = screen.getByTestId("button-test-up");
+    const counter = screen.getByTestId("counter-test")
+    
+    await user.click(powerDownButton)
+    await user.click(powerDownButton)
+    await user.click(powerUpButton)
+    
+    expect(counter.textContent).toEqual("-1")
+    
+  });
```
Comment (yulingyou) [inline Frontend/tests/components/buttons.test.jsx:69]:
```suggestion
  });
 test("Test that When User clicks  Power Up Button once and Power Down Button Twice, number changes to -1", async () =>  {
    const user = userEvent.setup()
    render(<Buttons />);
    const powerDownButton = screen.getByTestId("button-test-down");
    const powerUpButton = screen.getByTestId("button-test-up");
    const counter = screen.getByTestId("counter-test")

    await user.click(powerUpButton)
    await user.click(powerDownButton)
    await user.click(powerDownButton)
    

    expect(counter.textContent).toEqual("-1")

  });
```

File: Frontend/tests/components/buttons.test.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    const powerUpButton = screen.getByTestId("button-test-up");
+    const counter = screen.getByTestId("counter-test")
+    
+    await user.click(powerUpButton)
+    await user.click(powerUpButton)
+    await user.click(powerDownButton)
+    
+    expect(counter.textContent).toEqual("1")
+    
+  });
```
Comment (yulingyou) [inline Frontend/tests/components/buttons.test.jsx:98]:
```suggestion
  });
    test("Test that When User clicks Power Down Button once and Power Up Button Twice , Counter number changes to 1", async () =>  {
    const user = userEvent.setup()
    render(<Buttons />);
    const powerDownButton = screen.getByTestId("button-test-down");
    const powerUpButton = screen.getByTestId("button-test-up");
    const counter = screen.getByTestId("counter-test")

    await user.click(powerDownButton)
    await user.click(powerUpButton)
    await user.click(powerUpButton)
    

    expect(counter.textContent).toEqual("1")

  });
```

**Review by yulingyou — DISMISSED** [review]

**Review by Seyi-Toluhi — DISMISSED** [review]

**Review by tomgame984 — APPROVED** [review]

**Review by Seyi-Toluhi — APPROVED** [review]

**Review by JackCollett — APPROVED** [review]

**Pull request merged**